packages/node/test/unit/dev.test.ts

```diff
@@ -18,7 +18,56 @@ function testForkDevServer(entrypoint: string) {
     tsConfig: undefined,
     workPath: resolve(__dirname, '../dev-fixtures'),
     entrypoint,
-    devServerPath: resolve(__dirname, '../../dist/dev-server.js'),
+    devServerPath: resolve(__dirname, '../../dist/dev-server.mjs'),
+  });
+}
+
+interface FetchResponse {
+  headers: http.IncomingHttpHeaders;
+  json: () => Promise<any>;
+  status: number | undefined;
+  text: string;
+}
+
+/**
+ * This is a minimal replacement for `node-fetch`. We cannot use the real
+ * `node-fetch` because it is a ESM package and Jest doesn't support
+ * statically or dynamically importing ESM packages.
```

Comment:
Does Jest not support importing ESM packages? Or just the version of Jest we're using?

Reply:
The latest version of Jest has "experimental" support for ESM source files. Our code must be ESM to import ESM packages.

Jest supports TypeScript via either Babel or `ts-jest`. We use `ts-jest`. We specify the transpilation settings in the `tsconfig.json`, that alone isn't enough. We need to make sure we're using a newer version of `ts-jest` which supports `.mts` file extensions.

We default all `.js` and `.ts` files to use CommonJS and thus those file extensiosn can never load ESM packages such as `node-fetch`. We can only load ESM packages if we set the `"type": "module"` in the `package.json`, which we can't do yet, or we use `.mjs` and `.mts` file extensions.

Given this test file is a `.ts` file, it cannot load the ESM package `node-fetch`. As a quick workaround, I mock `fetch()` using Node.js's built-in `http` module. Once we drop support for Node.js 16 and older, we can use Node 18's built-in `fetch()`.

Reply:
One solution here is to set the tsconfig.json for jest to use `module: nodenext` and then you can import ESM packages just fine. I think you can do that here:

https://github.com/vercel/vercel/blob/66fe3db888ff1d736023ce7cb70e8cdd7ff69ed3/packages/node/test/tsconfig.json#L1-L4

Reply:
I just tested `module: 'nodenext'` and it had no effect. I tried renaming `dev.test.ts` to `.mts` as well as set the target and libs to ES2020 and ts-jest to `useESM`. Nothing works. I like Jest a little less now.

Reply:
Oh, I also tested `NODE_OPTIONS=--experimental-vm-modules` with no luck.

Reply:
How about reverting `"node-fetch": "3.3.1"` back to the previous version? Why was it updated to begin with?

Reply:
Reverted. Good question. I don't recall the exact reason for the update, but I recall running into issues at one point with importing the "default" for certain packages.

Reply:
This -> https://github.com/vercel/vercel/actions/runs/4681143008/jobs/8293359158?pr=9692#step:6:354